### DISCUSSION TITLE
4.1.0rc3

### DISCUSSION THREAD

**Discussion by sadpandajoe** [apache/superset#30631] [General]
Hello Superset Community,

This is a call for the vote to release Apache Superset version 4.1.0.

The release candidate:
https://dist.apache.org/repos/dist/dev/superset/4.1.0rc3/

The Git tag for the release:
https://github.com/apache/superset/tree/4.1.0rc3

The CHANGELOG for the release:
https://github.com/apache/superset/blob/4.1.0rc3/CHANGELOG/4.1.0.md

The instructions for updating to the release:
https://github.com/apache/superset/blob/4.1.0rc3/UPDATING.md

Public keys are available at:
https://www.apache.org/dist/superset/KEYS

<hr /><em>This discussion was created from the release <a href='https://github.com/apache/superset/releases/tag/4.1.0rc3'>4.1.0rc3</a>.</em>

**Comment by gpchandran** [comment]
Hello ,
My smoke testing looks good with 4.1.rc3, when you're planning 4.1.0 release tentatively?

**Reply by sadpandajoe** [reply]
Following the releasing page:
> Once 3+ binding votes (by PMC members) have been cast and at least 72 hours have past

so could be as early as Monday or could be later depending on how things are voted.

**Comment by jacob-roldan** [comment]
Is there an error to get the PDF or image in the latest version??

https://github.com/apache/superset/issues/30645

**Reply by gpchandran** [reply]
![image](https://github.com/user-attachments/assets/6ae450cc-57f5-4427-a54d-56e0747a7b34)
yes, seems like a bug, keep getting this message, image/pdf not getting downloaded

**Reply by geido** [reply]
> yes, seems like a bug, keep getting this message, image/pdf not getting downloaded

The message is suggesting to wait. Have you got an actual failure?

**Reply by gpchandran** [reply]
waited for few minutes and didn't get the download of image/pdf

**Reply by geido** [reply]
When the download fails, you will get a failure message. You should also check your resource allocation as this process is quite computationally expensive.

**Reply by gpchandran** [reply]
Hello ,
getting 404 not found error on below API calls
https://****/api/v1/dashboard/46/screenshot/c15f73c3387df8f2583bd50b1f4a0dc4/?download_format=pdf
https://****/api/v1/dashboard/46/screenshot/c15f73c3387df8f2583bd50b1f4a0dc4/?download_format=png


is this functionality working for you or other? not sure it's my environment or Superset application issue. I'm sure it's working for me on previous version 4.0.2

**Reply by gpchandran** [reply]
hi @geido - is this PDF/Image download works for you?

**Reply by sadpandajoe** [reply]
@gpchandran I believe @geido has a PR out (linked to the issue) that will make download work as it did before by moving the new logic behind a feature flag.

**Reply by geido** [reply]
@gpchandran you must have cache and Celery configured for this to work. If you have, then you should wait for the task to finish up and the endpoint will return 404 until the image is ready to be pulled from the cache.

**Reply by gpchandran** [reply]
Thanks a lot @sadpandajoe  and @geido - will this PR go in to 4.1.0rc4?
I've both Redis Cache and Celery running on our deployment.
ENABLE_DASHBOARD_DOWNLOAD_WEBDRIVER_SCREENSHOT - is enabling this feature flag mandatory  for download PDF/Image feature?

**Comment by padbk** [comment]
Am trying to test with python 3.11, but found no image on DockerHub https://github.com/apache/superset/issues/30672

**Reply by padbk** [reply]
This is merged now. https://github.com/apache/superset/pull/30733

Will it be possible to get this 🍒 'd so we get 4.1.0 python 3.11 images on Docker Hub please?

**Reply by sadpandajoe** [reply]
just cherried it, so should be part of rc4

**Comment by hscabral** [comment]
Is anyone having issues with `ModuleNotFoundError: No module named 'psycopg2'`? It does not happen in 4.0.2.

**Reply by gpchandran** [reply]
I faced same issue and added to this lib psycopg2-binary requirements.txt  to resolve it

**Reply by sfirke** [reply]
Yes this is due to a change in how docker images are built.

- Report of this issue on the RC2 thread: https://github.com/apache/superset/discussions/29999#discussioncomment-10490167
- Resolution: added a note to the 4.1.0 release notes https://github.com/apache/superset/blob/4.1.0rc3/RELEASING/release-notes-4-1/README.md#release-notes-for-superset-410
- I wonder if it should also be added to the UPDATING notes? https://github.com/apache/superset/blob/4.1.0rc3/UPDATING.md

**Reply by hscabral** [reply]
Building from scratch or running locally is fine. My use case involves downloading the images from Docker Hub to be used in a Kubernetes cluster. In RC3 I was able to make it work downloading the `-dev` image, but not sure if ideal to what we are doing. The issue seems to be widely reported, so I think I will follow the discussions and progress. In the release note it's fairly outlined too, as I noticed, thanks!


> Starting in 4.1.0, the release's docker image does not ship with drivers needed to operate Superset. Users may need to install a driver for their metadata database (MySQL or Postgres) as well as the driver for their data warehouse. This is a result of changes to the lean docker image that official releases come from; see [Docker Build Presets](https://github.com/apache/superset/blob/4.1.0rc3/docs/installation/docker-builds#build-presets) for more details.

**Reply by sfirke** [reply]
@hscabral I think running `-dev` is an okay approach for now.  Ideally there would be a pre-built image available that has some parts of `-dev` like the headless browser for screenshots and the psycopg2 driver but unlike `-dev` doesn't run as the root user... if someone wanted to share a Dockerfile for this I could work in December on making that part of the official build process.

**Comment by vicky-sai** [comment]
Any idea when are we planning for the release ?!
I am wondering the Schema specific access in SQL Lab will work only with this error fix.!

**Reply by sadpandajoe** [reply]
@vicky-sai did you see if this is resolved in 4.1.0rc3? If not do you have an open issue?

**Reply by vicky-sai** [reply]
@sadpandajoe 
Error: https://github.com/apache/superset/issues/28596
PR: https://github.com/apache/superset/pull/29311

This is something which I am awaiting to be fixed..

**Reply by sadpandajoe** [reply]
@vicky-sai did you see the comment in that PR? Someone mentioned it was fixed with the 4.1.0rc2, have you verified it isn't working? As for the PR you linked, in it's current state it's probably not mergeable as it's failing CI and could break for some databases.

**Comment by gpchandran** [comment]
Hi, getting this issue in 4.1.0rc3 (this was working fine in 4.0.2), unable to add Custom SQL for tabular chart Column.
Any one facing this issue, is this a bug or new validation added in 4.1? please advise

![image](https://github.com/user-attachments/assets/66202d27-e282-47ee-962d-d4efe3eca5d2)

**Reply by gpchandran** [reply]
Enabling this ALLOW_ADHOC_SUBQUERY feature flag fixes this issue

**Comment by sadpandajoe** [comment]
Voting is done and this release did no pass
